src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -151,6 +152,80 @@ sourceSets {
         buildResult.output.contains(':checkScalafixIntegTest SKIPPED')
     }
 
+    def 'scalafix semantic rule RemoveUnused fixes scala code'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FIXED_SCALA_CODE
+    }
+
+    def 'checkScalafix semantic rule RemoveUnused fails a build'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+
+        when:
+        runGradleTask('checkScalafix', [ ])
+
+        then:
+        // code is not changed on disk
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+        final UnexpectedBuildFailure err = thrown()
+        err.message.contains("A file on disk does not match the file contents if it was fixed with Scalafix")
+    }
+
+
+    def 'scalafix semantic rule RemoveUnused skips excluded scala files'() {
+        given:
+        scalafixConfFile.write "rules = [ RemoveUnused ]"
+        buildFile.append '''
+scalafix {
+  excludes = ["**/scalafix/**"]
+}
+'''
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+    }
+
+    def 'scalafix syntactic rule DisableSyntax.noVars fails a build'() {
+        given:
+        scalafixConfFile.write '''
+rules = [ DisableSyntax ]
+DisableSyntax.noVars = true
+'''
+
+        when:
+        runGradleTask('scalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+        final UnexpectedBuildFailure err = thrown()
+        err.message.contains("error: [DisableSyntax.var] mutable state should be avoided")
+    }
+
+    def 'checkScalafix syntactic rule DisableSyntax.noVars fails a build'() {
+        given:
+        scalafixConfFile.write '''
+rules = [ DisableSyntax ]
+DisableSyntax.noVars = true
+'''
+
+        when:
+        runGradleTask('checkScalafix', [ ])
+
+        then:
+        scalaSrcFile.getText() == ScalafixTestConstants.FLAWED_SCALA_CODE
+        final UnexpectedBuildFailure err = thrown()
+        err.message.contains("error: [DisableSyntax.var] mutable state should be avoided")
+    }
+
+
```

Comment:
it would be nice to have a test for the happy path: a source file without any issues :)